Sources/ProjectSpec/Target.swift

```diff
@@ -159,13 +162,17 @@ extension Target {
         guard let targetsDictionary: [String: JSONDictionary] = jsonDictionary["targets"] as? [String: JSONDictionary] else {
             return jsonDictionary
         }
-
+        
         var crossPlatformTargets: [String: JSONDictionary] = [:]
 
         for (targetName, target) in targetsDictionary {
-
+            var target = target
+            
+            if let platforms = target["platform"] as? [String], target["supportedPlatforms"] != nil {
+                target["platform"] = platforms.first
```

Comment:
This may be surprising. Could you leave a comment on the intention here?

Reply:
Basically if you are using `supportedPlatforms` and you want to support multiple platforms in the same target we can not allow xcodegen to generate multiple target for each platform if you use `platform` as an array.

So if you by mistake are adding both xcodegen ignores the array in `platform` taking as base platform the first one and it applies the new rules to support multiple platform in the same target.

Reply:
I think that people, now, can not use anymore version less than Xcode 14 to upload builds.

So one possibility could renaming `platform` in `basePlatform` or `sdkPlatform` (or something else) and completely remove the possibility to manage it as an array and generate different target per platform.

Reply:
Maybe throwing an error is more appropriate? Might make it clearer.  Saying something like you can't have an array of platforms if you define supportedPlatforms

Reply:
Yes, I will do that